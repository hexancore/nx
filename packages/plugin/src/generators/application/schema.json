{
  "$schema": "https://json-schema.org/schema",
  "$id": "Application",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the application is placed.",
      "x-priority": "important",
      "$default": {
        "$source": "argv",
        "index": 0
      },
    },
    "type": {
      "description": "The application type.",
      "type": "string",
      "enum": ["backend"],
      "x-prompt": "Which application type would you like to use ?",
      "x-priority": "important"
    }
  },
  "required": ["name", "type"]
}
