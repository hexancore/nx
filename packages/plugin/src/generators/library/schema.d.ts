import type { LibraryType } from "./LibraryType";

export interface LibraryGeneratorSchema {
  directory: string;
  type: LibraryType | 'backend' | 'frontend' | 'shared';
}