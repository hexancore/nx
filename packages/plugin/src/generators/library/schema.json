{
  "$schema": "https://json-schema.org/draft/2020-12/output/schema",
  "$id": "Library",
  "title": "",
  "type": "object",
  "properties": {
    "directory": {
      "type": "string",
      "description": "A directory where the library is placed in `apps/`. Project name will be directory with replaced '/'",
      "x-prompt": "Library project directory in `apps/`",
      "x-priority": "important",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "type": {
      "description": "The library type.",
      "type": "string",
      "enum": ["backend", "shared", "frontend"],
      "x-prompt": "Which library type would you like to use ?",
      "x-priority": "important"
    }
  },
  "required": ["directory", "type"]
}
