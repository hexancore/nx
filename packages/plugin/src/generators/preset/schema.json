{
  "$schema": "https://json-schema.org/draft/2020-12/output/schema",
  "$id": "Preset",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    }
  },
  "required": ["name"]
}
