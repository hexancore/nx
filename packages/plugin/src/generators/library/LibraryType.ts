export type LibraryType = 'frontend' | 'backend' | 'shared';