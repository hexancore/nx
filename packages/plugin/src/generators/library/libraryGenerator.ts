import {
  Tree,
  updateJson
} from '@nx/devkit';
import * as path from 'path';
import { HcNxHelper } from '../../util/HcNxHelper';
import { ProjectMeta, ProjectPackageJsonGenerator } from '../../util';
import { LibraryGeneratorSchema } from './schema';

export async function libraryGenerator(tree: Tree, options: LibraryGeneratorSchema): Promise<void> {

  const helper = new HcNxHelper(tree);
  const project: ProjectMeta = helper.createProjectMeta(options.directory, false);

  helper.addProjectConfiguration(project);
  ProjectPackageJsonGenerator.run(tree, { project });

  const filesPath = path.join(__dirname, 'files');
  helper.generateProjectFiles(project, path.join(filesPath, 'common'), '');

  updateJson(helper.tree, 'tsconfig.base.json', (v) => {
    v.compilerOptions.paths = v.paths ?? {};
    v.compilerOptions.paths[project.importName] = [`${project.root}/src/index.ts`];
    return v;
  });

  tree.write(path.join(project.root, 'src/index.ts'), 'export const INITIAL = 1;');
  tree.write(path.join(project.root, 'test/helper/.gitkeep'), '');
  tree.write(path.join(project.root, 'test/unit/.gitkeep'), '');
  tree.write(path.join(project.root, 'test/integration/.gitkeep'), '');
  helper.generateProjectFiles(project, path.join(filesPath, 'test'), 'test/unit');
}

export default libraryGenerator;
