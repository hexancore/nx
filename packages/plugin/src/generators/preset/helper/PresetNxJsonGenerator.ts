import {
  Tree,
  updateJson
} from '@nx/devkit';

export class PresetNxJsonGenerator {

  public static run(tree: Tree) {
    const g = new this();
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    updateJson(tree, 'nx.json', (o) => {
      o.extends = 'nx/presets/npm.json';
      o.pluginsConfig = {
        "@nx/js": {
          "analyzeSourceFiles": true
        }
      };

      o.namedInputs = { ...(o.namedInputs ?? {}), ...this.namedInputs() };
      o.targetDefaults = { ...(o.targetDefaults ?? {}), ...this.targetDefaults() };
      o.generators = {
        "@nx/workspace:move": {
          "projectNameAndRootFormat": "as-provided"
        }
      };
      o.useInferencePlugins = false;
      o.defaultBase = 'main';

      o['workspaceLayout'] = {
        "appsDir": "apps",
        "libsDir": "libs"
      };

      return o;
    });
  }

  private namedInputs(): Record<string, any> {
    return {
      "default": [
        "{projectRoot}/**/*",
        "!{projectRoot}/README.md",
        "sharedGlobals"
      ],
      "sharedGlobals": [],
      production: [
        "default",
        "!{projectRoot}/test/**/*",
        "!{projectRoot}/.vite/**/*",
        "!{projectRoot}/.storybook/**/*",
        "!{projectRoot}/jest.config.{js,ts,mjs,mts}",
        "!{projectRoot}/tsconfig.test.json",
        "!{projectRoot}/.eslintrc.json"
      ]
    };
  }

  private targetDefaults(): Record<string, any> {
    return {
      "@nx/js:tsc": {
        "inputs": ["production", "^production"],
        "outputs": ["{options.outputPath}"],
        "dependsOn": ["^build", "test"],
        "cache": true,
        "options": {
          "outputPath": "dist/{projectRoot}",
          "tsConfig": "{projectRoot}/tsconfig.build.json"
        }
      },

      "@nx/jest:jest": {
        "cache": true,
        "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
        "options": {
          "jestConfig": "{projectRoot}/jest.config.ts",
          "runInBand": true,
          "passWithNoTests": true
        },
        "configurations": {
          "watch": {
            "watch": true
          },
          "ci": {
            "ci": true,
            "codeCoverage": true
          },
          "updateSnapshot": {
            "updateSnapshot": true
          },
          "codeCoverage": {
            codeCoverage: true,
          }
        }
      },
      "@nx/eslint:lint": {
        "inputs": ["default", "{workspaceRoot}/.eslintrc.json", "{workspaceRoot}/.eslintignore"],
        "cache": true,
        "options": {
          "lintFilePatterns": ["{projectRoot}/**/*.ts", "{projectRoot}/package.json"]
        },
        "configurations": {
          "fix": {
            "fix": true,
            "cache": false
          }
        }
      },
      ...this.viteTargetsDefaults(),
      ...this.storybookTargetsDefaults(),
    };
  }

  private viteTargetsDefaults(): Record<string, any> {
    return {
      "@nx/vite:build": {
        "inputs": ["production", "^production"],
        "outputs": ["{options.outputPath}"],
        "defaultConfiguration": "production",
        "options": {
          "outputPath": "dist/{projectRoot}",
          "skipTypeCheck": true,
          "configFile": "{projectRoot}/vite.config.ts",
          "tsConfig": "{projectRoot}/tsconfig.build.json",
          "generatePackageJson": false
        },
        "configurations": {
          "development": {
            "mode": "development"
          },
          "production": {
            "mode": "production"
          }
        }
      },
      "@nx/vite:preview-server": {
        "defaultConfiguration": "development",
        "options": {
          "buildTarget": "build"
        },
        "configurations": {
          "development": {
            "buildTarget": "build:development"
          },
          "production": {
            "buildTarget": "build:production"
          }
        }
      },
      "@nx/vite:test": {
        "cache": true,
        "inputs": ["default", "^default"],
        "options": {
          "configFile": "{projectRoot}/vite.config.ts",
          "passWithNoTests": true,
          "reportsDirectory": "../../../coverage/{projectRoot}"
        },
        "configurations": {
          "watch": {
            "watch": true
          }
        }
      },
    };
  }

  private storybookTargetsDefaults(): Record<string, any> {
    return {
      "@nx/storybook:storybook": {
        "options": {
          "port": 4400,
          "configDir": "libs/front/mao1/.storybook"
        },
        "configurations": {
          "ci": {
            "quiet": true
          }
        }
      },
      "@nx/storybook:build": {
        "outputs": ["{options.outputDir}"],
        "options": {
          "outputDir": "dist/storybook/{projectRoot}",
          "configDir": "{projectRoot}/.storybook"
        },
        "configurations": {
          "ci": {
            "quiet": true
          }
        }
      },
      "storybook-test": {
        "executor": "nx:run-commands",
        "options": {
          "command": "test-storybook -c {projectRoot}/.storybook --url=http://localhost:4400"
        }
      },
      "storybook-static": {
        "executor": "@nx/web:file-server",
        "options": {
          "buildTarget": "storybook-build",
          "staticFilePath": "dist/storybook/{projectRoot}",
          "spa": true
        },
        "configurations": {
          "ci": {
            "buildTarget": "storybook-build:ci"
          }
        }
      }
    };
  }
}