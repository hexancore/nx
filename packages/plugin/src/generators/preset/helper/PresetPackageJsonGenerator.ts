import {
  addDependenciesToPackageJson,
  updateJson,
  Tree,
} from '@nx/devkit';

export class PresetPackageJsonGenerator {

  public constructor(
    private nxVersion: string,
    private hcNxPluginVersion: string
  ) {

  }

  public static run(tree: Tree, options: {
    nxVersion: string,
    hcNxPluginVersion: string
  }) {
    const g = new this(options.nxVersion, options.hcNxPluginVersion);
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    this.generateDeps(tree);

    updateJson(tree, 'package.json', (o) => {
      o.license = 'UNLICENSED';
      o.scripts = o.scripts ?? {};

      o.scripts['prepare'] = "node .husky/install.mjs",
        o.scripts['precommit'] = "nx run-many --targets=lint,test,build";
      o.scripts['ci'] = "nx run-many --targets=lint,test,build,e2e --configuration=ci";
      return o;
    });
  }

  private generateDeps(tree: Tree): void {
    const deps = this.createDeps();
    const devDeps = this.createDevDeps();
    addDependenciesToPackageJson(tree, deps, devDeps);
  }

  private createDeps(): Record<string, string> {
    const hexancore = {
      '@hexancore/core': '0.14.*',
      '@hexancore/common': '0.13.*',
      '@hexancore/cloud': '0.1.*',
      '@hexancore/auth': '0.3.*',
      '@hexancore/typeorm': '0.14.*',
    };

    return {
      ...hexancore,
      'tslib': '2.6.2'
    };
  }

  private createDevDeps(): Record<string, string> {
    return {
      ...this.createHexancoreDevDeps(),
      ...this.createNxDevDeps(),
      ...this.createFrontendDevDeps(),
      ...this.createJestDevDeps(),
      ...this.createEslintDevDeps(),
      ...this.createEssentialsDevDeps()
    };
  }

  private createNxDevDeps(): Record<string, string> {
    return {
      "@nx/devkit": this.nxVersion,
      "@nx/esbuild": this.nxVersion,
      "@nx/eslint-plugin": this.nxVersion,
      "@nx/jest": this.nxVersion,
      "@nx/js": this.nxVersion,
      "@nx/linter": this.nxVersion,
      "@nx/plugin": this.nxVersion,
    };
  }

  private createHexancoreDevDeps(): Record<string, string> {
    const [major, minor] = this.hcNxPluginVersion.split('.');

    return {
      "@hexancore/nx": `${major}.${minor}.*`,
      "@hexancore/mocker": "^1.1.2",
    };
  }

  private createFrontendDevDeps(): Record<string, string> {
    const vue = {
      "@vue/eslint-config-typescript": "^11.0.3",
      "@vue/test-utils": "^2.4.5",
      "@vue/tsconfig": "^0.4.0",
      "vue": "^3.4.25",
      "vue-eslint-parser": "^9.4.2",
      "vue-router": "^4.3.2",
      "pinia": "^2.1.7",
      "vue-tsc": "^1.8.27",
      "vee-validate": "^4.12.6",
    };

    const vueI18n = {
      "vue-i18n": "^9.13.1",
      "@intlify/eslint-plugin-vue-i18n": "^2.0.0",
      "@intlify/unplugin-vue-i18n": "^2.0.0",
      "unplugin-swc": "^1.4.5",
    };

    const vite = {
      "@vitejs/plugin-basic-ssl": "^1.1.0",
      "@vitejs/plugin-vue": "^5.0.4",
      "@vitest/coverage-v8": "1.6.*",
      "@vitest/ui": "1.6.*",
      "vitest": "1.6.*",
      "@rollup/plugin-node-resolve": "^15.2.3",
      "rollup": "^4.17.0",
      "vite": "^5.2.10",
      "vite-tsconfig-paths": "^4.3.2",
      "autoprefixer": "^10.4.19",
      "@babel/core": "^7.24.4",
      "@babel/plugin-transform-modules-commonjs": "^7.24.1",
      "@babel/preset-env": "^7.24.4",
    };

    const primevue = {
      "primeflex": "^3.3.1",
      "primeicons": "^6.0.1",
      "primevue": "^3.52.0",
    };

    const style = {
      "postcss": "^8.4.38",
      "sass": "^1.75.0",
      "stylelint": "^15.11.0",
      "stylelint-config-standard": "^33.0.0",
      "@samatech/postcss-basics": "^0.6.0",
    };

    return {
      ...vue,
      ...vueI18n,
      ...vite,
      ...primevue,
      ...style
    };
  }

  public createEslintDevDeps(): Record<string, string> {
    return {
      "eslint": "8.57.0",
      "eslint-plugin-import": "^2.29.1",
      "eslint-plugin-vue": "^9.25.0",
      "@typescript-eslint/parser": "7.3.0",
      "@stylistic/eslint-plugin": "^2.1.0",
    };
  }

  public createJestDevDeps(): Record<string, string> {
    return {
      "happy-dom": "^9.20.3",
      "jest": "^29.7.0",
      "jest-docblock": "^29.7.0",
      "jest-environment-jsdom": "^29.7.0",
      "jest-environment-node": "^29.7.0",
      "jest-expect-message": "^1.1.3",
      "jest-runner": "^29.7.0",
      "jest-runner-groups": "^2.2.0",
      "jsdom": "~22.1.0",
      "ts-jest": "^29.1.2",
      "@types/jest": "^29.5.12",
      "@jest/globals": "^29.7.0",
    };
  }

  public createEssentialsDevDeps(): Record<string, string> {
    return {
      "tsconfig": "^7.0.0",
      "typescript": "5.4.5",
      "typescript-transform-paths": "^3.4.7",
      "esbuild": "^0.19.12",
      "axios-mock-adapter": "^1.22.0",
      "@types/node": "20.12.12",
      "@nestjs/testing": "^10.3.8",
      "husky": "^9.0.11",
    };
  }
}