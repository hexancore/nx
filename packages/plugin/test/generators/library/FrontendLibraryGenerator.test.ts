import { Tree } from '@nx/devkit';

import { libraryGenerator } from '../../../src/generators/library/libraryGenerator';
import { LibraryGeneratorSchema } from '../../../src/generators/library/schema';
import { createEmptyNxTree, exceptOnlyFilesFromListExistInWorkspace, exceptWorkspaceFileMatchSnapshot } from '../../helper/functions';

describe('Library Generator: Frontend', () => {
  let tree: Tree;
  const type = 'frontend';
  const projectDirectory = `task-manager/${type}`;
  const projectRoot = `libs/${projectDirectory}`;
  beforeAll(async () => {
    tree = createEmptyNxTree();
    const options: LibraryGeneratorSchema = { directory: projectDirectory, type };
    await libraryGenerator(tree, options);
  });

  const expectedProjectFilesToMatchSnapshots = [
    'README.md',
    'project.json',
    'package.json',
    '.eslintrc.json',
    'tsconfig.json',
    'tsconfig.test.json',
    'tsconfig.build.json',
    'vite.config.mts',
    '.storybook/main.ts',
    '.storybook/preview.ts',
    'tsconfig.storybook.json',
    'src/vue-shims.d.ts',
    'src/Asset/Style/variables.scss',
    'src/Component/Button/Button.vue',
    'src/Component/Button/Button.stories.ts',
  ].map(f => `${projectRoot}/` + f);

  const expectedWorkspaceFilesToMatchSnapshots = [
    'tsconfig.base.json',
    'nx.json',
    'package.json',
    ...expectedProjectFilesToMatchSnapshots
  ];

  test.each(expectedWorkspaceFilesToMatchSnapshots.map(i => [i]))('%s should match snapshot', (filePath) => {
    exceptWorkspaceFileMatchSnapshot(tree, filePath);
  });

  test('only file from list should exist', () => {
    const expected = [
      '.prettierrc',
      `${projectRoot}/src/Asset/Image/logo.png`,
      `${projectRoot}/test/config.ts`,
      `${projectRoot}/test/unit/sample.test.ts`,
      `${projectRoot}/test/helper/.gitkeep`,
      `${projectRoot}/test/integration/.gitkeep`,
      `${projectRoot}/src/Service/.gitkeep`,
      ...expectedWorkspaceFilesToMatchSnapshots
    ];
    exceptOnlyFilesFromListExistInWorkspace(tree, expected);
  });
});
