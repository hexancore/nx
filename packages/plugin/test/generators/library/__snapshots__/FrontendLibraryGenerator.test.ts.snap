// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library Generator: Frontend libs/task-manager/frontend/.eslintrc.json should match snapshot 1`] = `
"{
  "extends": [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier/skip-formatting",
    "../../../.eslintrc.json"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx", "*.vue"],
      "rules": {
        "vue/multi-word-component-names": "off"
      }
    }
  ]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/.storybook/main.ts should match snapshot 1`] = `
"import { dirname } from "path";
import { defineProjectStorybookConfig } from "../../../../.workspace/storybook/storybook.preset";

export default defineProjectStorybookConfig({
  workspaceProjectDir: dirname(__dirname),
  staticDirs: [],
});
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/.storybook/preview.ts should match snapshot 1`] = `"export const tags = ["autodocs"];"`;

exports[`Library Generator: Frontend libs/task-manager/frontend/README.md should match snapshot 1`] = `
"# task-manager-frontend


"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/package.json should match snapshot 1`] = `
"{
  "name": "@testcompany/task-manager-frontend",
  "type": "module",
  "private": true,
  "dependencies": {
    "tslib": "2.6.2"
  },
  "sideEffects": [
    "**/*.css"
  ]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/project.json should match snapshot 1`] = `
"{
  "name": "task-manager-frontend",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/task-manager/frontend/src",
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/vite:test"
    },
    "build": {
      "executor": "@nx/vite:build"
    },
    "storybook": {
      "executor": "@nx/storybook:storybook"
    },
    "storybook-test": {
      "executor": "nx:run-commands"
    },
    "storybook-build": {
      "executor": "@nx/storybook:build"
    },
    "storybook-static": {
      "executor": "@nx/web:file-server"
    }
  }
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/src/Asset/Style/variables.scss should match snapshot 1`] = `
"$primary-color: #3498db;
$secondary-color: #2ecc71;

body {
  background-color: $primary-color;
}"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/src/Component/Button/Button.stories.ts should match snapshot 1`] = `
"import type { Meta, StoryObj } from '@storybook/vue3';

import Button from './Button.vue';
import { h } from 'vue';

const meta = {
  title: 'Example/Button',
  component: Button,

  tags: ['autodocs'],
  argTypes: {
    size: { control: 'select', options: ['small', 'medium', 'large'] },
    backgroundColor: { control: 'color' },
    onClick: { action: 'clicked' },
  },
  args: { primary: false }, // default value
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;
const iconImg = 'image/logo.png';

export const Primary: Story = {
  args: {
    primary: true,
    label: 'Hexancore',
    icon: h('img',{src:iconImg,height:64}),
    default:h('span',{style:{color:'yellow',padding:'16px'}},'Default Slot Content')
  },
};

export const Secondary: Story = {
  args: {
    primary: false,
    label: 'Hexancore',
    icon: h('img',{src:iconImg,height:64}),
    default:()=>'Default Slot Content as Text',
  },
};"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/src/Component/Button/Button.vue should match snapshot 1`] = `
"<template>
  <button type="button" :class="classes" @click="onClick" :style="style">
    <slot name="icon"></slot> {{ label }} <slot></slot>
    <slot name="BoxComponent"></slot>
  </button>

</template>

<script lang="ts" setup>
import { computed } from 'vue';

const props = withDefaults(defineProps<{
  /**
   * The label of the button
   */
  label: string,
  /**
   * primary or secondary button
   */
  primary?: boolean,
  /**
   * size of the button
   */
  size?: 'small' | 'medium' | 'large',
  /**
   * background color of the button
   */
  backgroundColor?: string,

}>(), { primary: false });

const emit = defineEmits<{
  (e: 'click', id: number): void;
}>();

const classes = computed(() => ({
  'storybook-button': true,
  'storybook-button--primary': props.primary,
  'storybook-button--secondary': !props.primary,
  [\`storybook-button--\${props.size || 'medium'}\`]: true,
}));

const style = computed(() => ({
  backgroundColor: props.backgroundColor
}));

const onClick = () => {
  emit("click", 1)
};

</script>

<style lang='scss'>

.storybook-button {
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 700;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  line-height: 1;
}

.storybook-button:hover {
    background-color: #007bb5;
    transform: scale(1.05);
}


.storybook-button:active {
    background-color: #005f87;
    transform: scale(0.95);
}

.storybook-button--primary {
  color: white;
  background-color: #1ea7fd;
}
.storybook-button--secondary {
  color: #333;
  background-color: transparent;
  box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
}

.storybook-button--secondary:hover {
  background-color: rgba(0, 0, 0, 0.1); /* Lekko ciemniejsze tło przy hover */
  box-shadow: rgba(0, 0, 0, 0.25) 0px 0px 0px 1px inset;
  transform: scale(1.05);
}

.storybook-button--secondary:active {
  background-color: rgba(0, 0, 0, 0.2); /* Jeszcze ciemniejsze tło przy aktywacji */
  box-shadow: rgba(0, 0, 0, 0.35) 0px 0px 0px 1px inset;
  transform: scale(0.95);
}

.storybook-button--small {
  font-size: 12px;
  padding: 10px 16px;
}
.storybook-button--medium {
  font-size: 14px;
  padding: 11px 20px;
}
.storybook-button--large {
  font-size: 16px;
  padding: 12px 24px;
}
</style>"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/src/vue-shims.d.ts should match snapshot 1`] = `
"declare module '*.vue' {
  import { defineComponent } from 'vue';
  const component: ReturnType<typeof defineComponent>;
  export default component;
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/tsconfig.build.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "types": ["vite/client"]
  },
  "exclude": [
    "**/*.stories.ts"
  ],
  "include": ["src/**/*.ts","src/**/*.tsx", "src/**/*.vue"]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/tsconfig.json should match snapshot 1`] = `
"{
  "extends": "../../../tsconfig.frontend.base.json",
  "compilerOptions": {
    "paths": {
      "@": [
        "libs/task-manager/frontend/src"
      ],
      "@/*": [
        "libs/task-manager/frontend/src/*"
      ],
      "@test/*": [
        "libs/task-manager/frontend/test/helper/*"
      ]
    }
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.build.json"
    },
    {
      "path": "./tsconfig.test.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/tsconfig.storybook.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true
  },
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.tsx",
    ".storybook/*.ts",
    ".storybook/*.mts"
  ]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/tsconfig.test.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "types": ["vitest/globals", "vitest/importMeta", "vite/client", "node", "vitest"]
  },
  "include": ["vite.config.mts", "test/**/*.ts", "test/**/*.tsx"]
}
"
`;

exports[`Library Generator: Frontend libs/task-manager/frontend/vite.config.mts should match snapshot 1`] = `
"import { defineLibProjectViteConfig } from "../../../.workspace/vite/vite.preset.mjs";

const options = {
  workspaceProjectDir: __dirname
};
export default defineLibProjectViteConfig(options);
"
`;

exports[`Library Generator: Frontend nx.json should match snapshot 1`] = `"{"affected":{"defaultBase":"main"},"targetDefaults":{"build":{"cache":true},"lint":{"cache":true}}}"`;

exports[`Library Generator: Frontend package.json should match snapshot 1`] = `
"{
  "name": "@testcompany/source"
}
"
`;

exports[`Library Generator: Frontend tsconfig.base.json should match snapshot 1`] = `
"{
  "compilerOptions": {
    "paths": {
      "@testcompany/task-manager-frontend": [
        "libs/task-manager/frontend/src/index.ts"
      ]
    }
  }
}
"
`;
